### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by mattbrannon**
This PR attempts to simplify the process of setting up a new project with the `vs-code-prettier-eslint` extension. Changes to the `README.md` include the following:

- A general restructuring of the document. 
    - The example projects and troubleshooting sections are moved to the end of the document. 
    - The installation and setup sections are moved to the beginning
- The sections detailing installation and setup and have been mostly rewritten.

Moving the troubleshooting and examples sections to the end and the installation section to the beginning allows the reader to read from top to bottom without as much jumping back and forth between sections.  

The goal of the rewrite is help people get things installed and setup as quickly and easily as possible. Code examples are provided that users can copy/paste directly into their terminal. The section detailing setup includes a json snippet that users can paste directly into their workspace settings. A link to the troubleshooting section is also provided in case something goes wrong. 

It's quite a big change to the current documentation so please let me know what you think. Any and all feedback is welcome.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,119 +20,114 @@ This extension requires the following NPM packages to be installed either locall
...
 
-## Installation
+## if your project uses typescript
+yarn add -D prettier@^2.5.1 eslint@^8.7.0 @typescript-eslint/parser@^5.0.1 typescript@^4.4.4
+```
 
-**Notes**:
+## Project settings 
 
-- The [Prettier extension](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) is not required.
```
Comment (idahogurl) [inline README.md:65]: This is actually important. You wouldn't think so but I had people thinking they had to have it installed.

File: README.md [hunk 2]
```diff
@@ -20,119 +20,114 @@ This extension requires the following NPM packages to be installed either locall
...
 
-**Note**: There are known performance issues with quad-core or slower processors. The slowness comes from ESLint CLI which the 'prettier-eslint' package because uses.
+```bash 
+## minimum requirements
+yarn add -D prettier@^2.5.1 eslint@^8.7.0
 
-## Installation
+## if your project uses typescript
+yarn add -D prettier@^2.5.1 eslint@^8.7.0 @typescript-eslint/parser@^5.0.1 typescript@^4.4.4
+```
```
Comment (idahogurl) [inline README.md:43]:
```suggestion
### TypeScript Projects
```bash
yarn add -D prettier@^2.5.1 eslint@^8.7.0 @typescript-eslint/parser@^5.0.1 typescript@^4.4.4
```

### Vue Projects
```bash
yarn add -D prettier@^2.5.1 eslint@^8.7.0 vue-eslint-parser@^8.0.0
```
```

File: README.md [hunk 3]
```diff
@@ -20,119 +20,114 @@ This extension requires the following NPM packages to be installed either locall
...
-8. If restarting did not work:
-   - If you saw _Error: Cannot find module '@typescript-eslint/parser'_, view the [TypeScript project](https://github.com/idahogurl/vs-code-prettier-eslint/tree/master/examples/typescript) for a working example.
-   - Make sure you have the required packages installed locally (global installations don't work sometimes)
+- The [Prettier extension](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) is **not** required.
 
-**Note**: Most issues are caused by using an unsupported ESLint version or an invalid ESLint configuration.
+- The [ESLint extension](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint) is **not** required. However, it is needed to have lint errors show while editing your file.
 
-Running your files through the [Prettier ESLint CLI](https://github.com/prettier/prettier-eslint-cli) first is a good way to determine if it's the extension, the `prettier-eslint` package, or your configuration. 
+## Install dependencies
```
Comment (idahogurl) [inline README.md:33]:
```suggestion
## Install Dependencies
```

File: README.md [hunk 4]
```diff
@@ -20,119 +20,114 @@ This extension requires the following NPM packages to be installed either locall
...
+   - Make sure you have the required packages installed locally (global installations don't work sometimes)
 
-(From https://eslint.org/docs/user-guide/configuring)
+**Note**: Most issues are caused by using an unsupported ESLint version or an invalid ESLint configuration.
 
-Use a JavaScript, JSON or YAML file to specify configuration information for an entire directory and all of its subdirectories. This can be in the form of an [`.eslintrc.*`](https://eslint.org/docs/user-guide/configuring#configuration-file-formats) file or an `eslintConfig` field in a [`package.json`](https://docs.npmjs.com/files/package.json) file.
+Running your files through the [Prettier ESLint CLI](https://github.com/prettier/prettier-eslint-cli) first is a good way to determine if it's the extension, the `prettier-eslint` package, or your configuration. 
 
-### Prettier Configuration File
+You can run `eslint --print-config .eslintrc.js` to check your ESLint configuration. This prints out the configuration being used after it applies plugins & extended configurations.
```
Comment (idahogurl) [inline README.md:122]:
```suggestion
You can run `npx eslint --print-config .eslintrc.js` to check your ESLint configuration. This prints out the configuration being used after it applies plugins & extended configurations.
```

Might as well make changes I've wanted to make for a while 🙂

File: README.md [hunk 5]
```diff
@@ -20,119 +20,114 @@ This extension requires the following NPM packages to be installed either locall
...
 
-(From https://prettier.io/docs/en/configuration.html)
+**Note**: There are known performance issues with quad-core or slower processors. The slowness comes from ESLint CLI which the 'prettier-eslint' package uses.
```
Comment (idahogurl) [inline README.md:124]:
```suggestion
**Note**: There are known performance issues with quad-core or slower processors. The slowness comes from the 'prettier-eslint' package the extension uses.
```

Might as well make changes I've wanted to make for a while 🙂

File: README.md [hunk 6]
```diff
@@ -20,119 +20,114 @@ This extension requires the following NPM packages to be installed either locall
...
+      "editor.formatOnSave": true, // optional 
+      "editor.formatOnSaveMode": "file", // required to format on save
+      "files.autoSave": "onFocusChange" // optional but recommended
+    }
+    ```
 
-1. Install the extension
-2. Quit VS Code
-3. Restart VS Code
+4. **Restart VsCode**
```
Comment (idahogurl) [inline README.md:68]:
```suggestion
4. **Restart VS Code**  
```

File: README.md [hunk 7]
```diff
@@ -20,119 +20,114 @@ This extension requires the following NPM packages to be installed either locall
...
+```
 
-**Notes**:
+## Project settings 
 
-- The [Prettier extension](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) is not required.
+Next we need to configure your project to use the extension. To do that, we're going to open or create a `settings.json` file at the root of your project. If you already have a `.vscode/settings.json` file in your project, you can skip the first two steps below and jump straight to step 3. 
 
-- The [ESLint extension](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint) is not required. However, it is needed to have lint errors show while editing your file.
+1. Open the command palette in vscode by typing:
```
Comment (idahogurl) [inline README.md:49]:
```suggestion
1. Open the command palette in VS Code by typing:
```

This is how Microsoft abbreviates Visual Studio Code. See screenshot below.
Screen Shot 2022-07-20 at 10 08 53 AM

**Review by idahogurl — CHANGES_REQUESTED** [review]
Thanks Matt for the PR! This does look better. The reasoning behind putting the troubleshooting and example projects first was that I got so many issues where people wouldn't even check the troubleshooting section before they created an issue. 

Could we put a TOC with some of the 2nd level headers as links (Installation, Configuration, Troubleshooting, Example Projects)

**Comment by mattbrannon** [comment]
@idahogurl Apologies for the late response. Those were all good suggestions.

**Review by idahogurl — APPROVED** [review]

**Comment by idahogurl** [comment]
Closes #46

**Comment by mattbrannon** [comment]
> Could we put a TOC with some of the 2nd level headers as links (Installation, Configuration, Troubleshooting, Example Projects)

@idahogurl I completely missed your request before submitting my final request for a review of changes. My apologies for that. I see that you implemented it yourself though. It looks great! Thanks for accepting the PR. I hope it helps reduce some of the complexity with installing and configuring everything. I love this extension. It's the default for all my personal projects. Thanks for making it and thanks for allowing me to contribute to it!

**Comment by idahogurl** [comment]
@mattbrannon No worries! You did most of the heavy lifting. Thanks for the PR again.

**Pull request merged**